core/frontend/src/tile/map/MapLayerSources.ts

```diff
@@ -112,12 +128,33 @@ export class MapLayerSource {
       layerSettings?.setCredentials(this.userName, this.password);
     }
 
+    if (this.savedQueryParams) {
+      layerSettings.savedQueryParams = {...this.savedQueryParams};
+    }
+
+    if (this.unsavedQueryParams) {
+      layerSettings.unsavedQueryParams = {...this.unsavedQueryParams};
+    }
     return layerSettings;
   }
 
   private getCredentials(): RequestBasicCredentials | undefined {
     return this.userName && this.password ? { user: this.userName, password: this.password } : undefined;
   }
+
+  /** Get all query parameters
+ * @beta
+ */
+  public get queryParameters() {
```

Comment:
This should probably be a method like `collectQueryParams()`. Getter makes it look like a simple property access which has performance implications but more importantly implies that repeated calls will return the same object. Some bundlers make that latter assumption to optimize code, which will produce incorrect results in this case.

Reply:
I agree, done.